test/library_tests.hpp

```diff
@@ -368,5 +378,247 @@ TEST_CASE("addWeakEdge", "[pathfinding][interface]") {
     REQUIRE(res[21] == true);
     REQUIRE(res[22] == true);
     REQUIRE(res[23] == true);
+
+    Pathfinding::NodeArray nodes = Pathfinding::getDebugNodes();
+
+    SECTION("Node 0 connected to Node 1") {
+        bool connected = false;
```

Comment:
I dont know if it is possible, but you may want to put the ```... connected to ...``` tests within a function, and call them with  corresponding arguments.